server/controllers/medical/patients/index.js

```diff
@@ -423,6 +427,233 @@ function searchByName(req, res, next) {
+            // console.log('DOBS: ', dob, dobYearOnly, patientDob, patientDobYearOnly);
+
+            if (dobYearOnly || patientDobYearOnly) {
+              // If either specified only with the year
+              // NOTE: Treating either as year-only the same way
+              const dobYear = dob.getYear();
+              const patientDobYear = patientDob.getYear();
+              if (dobYear === patientDobYear) {
+                // Full score if both years match and both are year-only
+                score += dobWeight * 1.0;
+                // console.log("Year match", dobYear, score);
+              } else {
+                // Downgrade the score by the number of years off
+                const maxYearsDiff = 5;
+                const yearsDiff = Math.abs(dobYear - patientDobYear);
+                if (yearsDiff <= maxYearsDiff) {
+                  // Discount year near matches proportionately
+                  score += dobWeight * 0.8 * (1.0 - yearsDiff / maxYearsDiff);
+                  // console.log("Near year match", yearsDiff, score);
+                }
+              }
+            } else {
+              // We have exact dates for both
+              const daysDiff = Math.round(Math.abs(dob - patientDob) / (1000 * 24 * 3600));
+              // console.log("DaysDiff: ",daysDiff);
+              if (daysDiff === 0) {
+                // Count the same day as best dob match
+                score += dobWeight * 1.0;
+                // console.log("Day match: ", score)
+              } else {
+                // Discount appropriately
+                const maxDaysDiff = 730; // 2 years
+                if (daysDiff <= maxDaysDiff) {
+                  score += dobWeight * 0.8 * (1.0 - daysDiff / maxDaysDiff);
+                  // console.log("Near day match: ", daysDiff, score);
+                }
+              }
+            }
+          }
+        }
+        // console.log("Score: ", score, maxScore);
+        matches.push([pid, score / maxScore]);
+      });
+
+      // If there are no matches, return now
+      if (matches.length === 0) {
+        return res.status(200).json([]);
+      }
+
+      // Resort the matches
+      matches = matches.sort((a, b) => { return b[1] - a[1]; });
+
+      // console.log("NameMatches: ");
+      // nameMatches.forEach(([pid, sname, score]) => {
+      //   console.log(pid, sname, patientNames[pid], score);
+      // });
+
+      // Now get the info for these patients
+      find({ uuids : matches.map(x => x[0]) })
+        .then((data) => {
+          // Insert the match score into each record
+          data.forEach((row) => {
+            const [/* name */, mscore] = matches.find(mrow => { return mrow[0] === row.uuid; });
+            row.matchScore = mscore;
+          });
+          return res.status(200).json(data);
+        })
+        .catch(next)
+        .done();
+
+      return res.status(200).json([]); // Keep Hound from complaining
```

Comment:
This is where your error is occurring.  You have already called res.status().send() above, in the .then() promise chain.  Since each HTTP request can only return one result, express is complaining as this .json(), will trigger the return, followed by the .send() from a few lines above it.

Reply:
Here is my proposal to clean this up and avoid sending two http responses: let's make this entire function an async function!  It will be easier to read as well.

Basically, use async in front of the function call, then use await in front of the db.exec(), and await in front of the find() call.  Then you can return the res.status().json() as the last call in the function without thinking about the .then() promise chain since it will be unraveled.

It should also make the code a bit easier to read.

Reply:
Until we can chat tomorrow, I did a hack on this to return [] to keep hound happy.  This line will never execute, so it is isn't a problem.  But I would like to fix it as you suggested.   Have not been able to get to it today.

Reply:
With help from @jniles I implemented this fix.